framework/src/controller/helpers/configurator/configurator.js

```diff
@@ -0,0 +1,148 @@
+const _ = require('lodash');
+const yargs = require('yargs');
+const { config: loggerConfig } = require('../../../components/logger/defaults');
+const { config: storageConfig } = require('../../../components/storage/defaults');
+const { config: cacheConfig } = require('../../../components/cache/defaults');
+const { applicationConfigSchema } = require('../../schema');
+
+const { parseEnvArgAndValidate } = require('./../validator');
+
+const sanitizeSchemaKeys = (keyString, extraKeyToRemove) =>
+	keyString.replace(new RegExp(`properties.|(${extraKeyToRemove})`, 'g'), '');
+
+const traverseObject = (o, func, parent = undefined) => {
+	Object.keys(o).forEach(key => {
+		func.apply(o, [key, o[key], parent]);
+		if (o[key] !== null && typeof o[key] === 'object') {
+			if (parent) {
+				traverseObject(o[key], func, `${parent}.${key}`);
+			} else {
+				traverseObject(o[key], func, key);
+			}
+		}
+	});
+};
+
+class Configurator {
+	constructor() {
+		this.configSchema = applicationConfigSchema;
+		this.metaInfo = {};
+		this.listOfArgs = new Set();
+
+		this.customData = [];
+
+		this.registerSchema(loggerConfig, 'components.logger');
```

Comment:
these register should happen outside of this imo

Reply:
Components were very generic part of the framework so I keep those there. Will move them to main file.